File: Assignment-1/Q5.cpp
class Solution {
public:
    void merge(vector<int>& n1, int m, vector<int>& n2, int n) {
        int p = m - 1;
        int q = n - 1;
        int j = m + n - 1;

        while (q >= 0 && p >= 0) {
            if (n1[p] < n2[q]) {
                n1[j] = n2[q];
                j--;
                q--;
            }
            else {
                n1[j] = n1[p];
                j--;
                p--;
            }
        }

        while (q >= 0) {
            n1[j] = n2[q];
            j--;
            q--;
        }
    }
};

File: Assignment 3/Q5.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if not digits:
            return [1]  # Return [1] if digits list is empty
        
        i = len(digits) - 1
        carry = 1
        
        while carry != 0 and i >= 0:  # Add a check for valid index
            if digits[i] == 9:
                digits[i] = 0
                carry = 1
            else:
                digits[i] += 1
                carry = 0
            i -= 1
        
        if carry == 1:  # If carry is still 1 after the loop, prepend 1 to the list
            digits.insert(0, 1)
        
        return digits